docs/src/pages/components/data-grid/pagination/CursorPaginationGrid.js

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { DataGrid } from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+
+const PAGE_SIZE = 5;
+
+function loadServerRows(cursor, data) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      const start = cursor ? data.cursors.indexOf(cursor) : 0;
+      const end = start + PAGE_SIZE;
+      const rows = data.rows.slice(start, end);
+
+      resolve({ rows, nextCursor: data.cursors[end] });
+    }, Math.random() * 500 + 100); // simulate network latency
+  });
+}
+
+const useCursorBasedDemoData = (options) => {
+  const response = useDemoData(options);
+
+  const data = React.useMemo(
+    () => ({
+      ...response.data,
+      cursors: response.data.rows.map(() => Math.random().toString()),
```

Comment:
We could `row.Id` as cursor

Reply:
https://slack.engineering/evolving-api-pagination-at-slack/

Reply:
Done :+1: